test/kes_dashboard_live_tiles_cli/TestTile.h

```diff
@@ -12,71 +12,340 @@
+ * does not match what is expected.
  */
-TEST(TestTile, SerialiseAlsoWritesCoverPath) {
-    // TODO
-    EXPECT_TRUE(false);
+TEST_F(KesDltResponsesFixture, InitialiseFailsForMissingKESContractMembers) {
+    JSON_Value* tileData =
+        json_array_get_value(
+            json_object_get_array(
+                json_value_get_object(
+                    this->responses["invalid_feed2"].get()),
+                "tiles"),
+            0);
+    JSON_Object* data = json_value_get_object(tileData);
+
+    Tile tile;
+
+    EXPECT_FALSE(tile.initialise(tileData));
 }
 
 /**
- * TODO
+ * Check that Tile.serialise() serialises all members pertinent to a valid
+ * KES response.
  */
-TEST(TestTile, DownloadSetsCoverPathWithId) {
-    // TODO
-    EXPECT_TRUE(false);
+TEST_F(KesDltResponsesFixture, SerialiseWritesAllKESContractMembers) {
+    JSON_Value* tileData =
+        json_array_get_value(
+            json_object_get_array(
+                json_value_get_object(
+                    this->responses["feed1"].get()),
+                "tiles"),
+            0);
+    JSON_Object* data = json_value_get_object(tileData);
+
+    Tile tile;
+    tile.initialise(tileData);
+
+    shared_ptr<JSON_Value> result(
+        tile.serialise(), json_value_free);
+
+    shared_ptr<char> resultStr(
+        json_serialize_to_string_pretty(result.get()),
+        json_free_serialized_string);
+    shared_ptr<char> tileDataStr(
+        json_serialize_to_string_pretty(tileData),
+        json_free_serialized_string);
+
+    EXPECT_THAT(result.get(), JsonEq(tileData))
+        << "tileData was: " << string(tileDataStr.get()) << endl
+        << "result was: " << string(resultStr.get()) << endl;
+}
+
+/**
+ * Check that Tile.serialise() serialises all members pertinent to a valid
+ * KES response and also coverPath when it's set.
+ */
+TEST_F(KesDltResponsesFixture, SerialiseAlsoWritesCoverPath) {
+    JSON_Value* tileData =
+        json_array_get_value(
+            json_object_get_array(
+                json_value_get_object(
+                    this->caches["tile_cache1"].get()),
+                "tiles"),
+            0);
+    JSON_Object* data = json_value_get_object(tileData);
+
+    Tile tile;
+    tile.initialise(tileData);
+
+    shared_ptr<JSON_Value> result(
+        tile.serialise(), json_value_free);
+
+    shared_ptr<char> resultStr(
+        json_serialize_to_string_pretty(result.get()),
+        json_free_serialized_string);
+    shared_ptr<char> tileDataStr(
+        json_serialize_to_string_pretty(tileData),
+        json_free_serialized_string);
+
+    EXPECT_THAT(result.get(), JsonEq(tileData))
+        << "tileData was: " << string(tileDataStr.get()) << endl
+        << "result was: " << string(resultStr.get()) << endl;
 }
 
 /**
  * TODO
  */
-TEST(TestTile, DownloadCreatesTileDirDirectories) {
+/*TEST(TestTile, DownloadCreatesTileDirDirectories) {
     // TODO
     EXPECT_TRUE(false);
+}*/
```

Comment:
You want to get rid of this?

Reply:
Clear reminder for me to come back to it